src/main/java/com/amazon/jenkins/ec2fleet/EC2FleetCloud.java

```diff
@@ -376,9 +377,9 @@ private void addNewSlave(final AmazonEC2 ec2, final String instanceId) throws Ex
         if (address == null)
             return; // Wait some more...
 
-        final FleetNode slave = new FleetNode(instanceId, "Fleet slave for" + instanceId,
+        final FleetNode slave = new FleetNode(instanceId, "Fleet slave for " + this.name + " (" + instanceId + ")",
```

Comment:
Instead of splitting name generation logic into two places here and ```FleetNode.getDisplayName``` could we move description inside ```FleetNode``` constructor